---
title: "Ovarian Samples Proteomics Data Analysis"
author: "TÃ¼lay Karakulak"
date: "2024-05-12"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#required libraries
```{r loadlibraries, echo=FALSE}
library(tidyverse)
library(tidyr)
library(dplyr)
library(ggplot2)
library(pheatmap)
library(sva)
library(stats)
library(data.table)
library(BSDA)
library(Biostrings)
library(ggvenn)
```

```{r read_output_diann}
Main_output <- read_delim('/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Proteomics/Ovary/DIANN/Main_Report.tsv', delim = '\t', col_names = TRUE)
head(Main_output)
Main_output <- Main_output %>% filter(!str_detect(Genes, ";"))
head(Main_output)
dim(Main_output)
```


```{r read_MDTs}
Switches_Ovary_pri <- readRDS('/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Ovary/Switches/CorrectedStat/Switches_TuPro_Pri_GTEx.RDS')
Switches_Ovary_met <- readRDS('/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Ovary/Switches/CorrectedStat/Switches_TuPro_Met_GTEx.RDS')
Switches_Ovary <- rbind(Switches_Ovary_pri, Switches_Ovary_met)
```


```{r merge_dMDTs_ENSPs}
ensg_enst_ensp_apris_v104 <-  read.csv('/Users/tulaykarakulak/Documents/PhD/Projects/GTExSuitibility/Ensembl/mart_export-2May2023.txt', sep='\t')
head(ensg_enst_ensp_apris_v104)
```

```{r merge_ensts}
Switches_Ovary_ensps <- merge(ensg_enst_ensp_apris_v104, Switches_Ovary, by.x= 'Transcript.stable.ID', by.y = 'dMDT', sort=FALSE)
head(Switches_Ovary_ensps)
```



```{r check_dMDT_GTEx}
# filter main output
Main_output_filtered <- Main_output[Main_output$Q.Value <= 0.01 & Main_output$PG.Q.Value <= 0.01, ]
# check dMDT in protein.id column

Main_output_filtered_dMDT <- Main_output_filtered[Main_output_filtered$Protein.Ids %in% Switches_Ovary_ensps$Protein.stable.ID, ]

head(Main_output_filtered_dMDT)
dim(Main_output_filtered_dMDT)
#[1] 7561   58

unique_peptides <- Main_output_filtered_dMDT[,c('Protein.Ids','Stripped.Sequence')] %>% dplyr::distinct()
```
> length(unique(Switches_Ovary_ensps$Transcript.stable.ID))
[1] 2054
> length(unique(Main_output_filtered_dMDT$Protein.Ids))
[1] 121
# 121/2054*100 : %5.890944 of dMDTs are identified with this way in Ovarian Sample. - it is not sample specific approach. All peptides identified in all cohort. 


```{r mdt_enst_merging}
Switches_Ovary_MDTS_splitted <- Switches_Ovary %>%
  mutate(MDT_GTEx = strsplit(as.character(MDT_GTEx), ",")) %>%
  unnest(MDT_GTEx)

Switches_Ovary_ensps_mdts <- merge(ensg_enst_ensp_apris_v104, Switches_Ovary_MDTS_splitted, by.x= 'Transcript.stable.ID', by.y = 'MDT_GTEx', sort=FALSE)
head(Switches_Ovary_ensps_mdts)
```


```{r upper_code_vectorizaition}
library(data.table)
dMDT_vector <- unique(Switches_Ovary_ensps$Protein.stable.ID)
MDT_vector <- unique(Switches_Ovary_ensps_mdts$Protein.stable.ID)

dMDT_proteomics <- Main_output_filtered[FALSE, ]
MDT_proteomics <- Main_output_filtered[FALSE, ]

# Convert data frames to data.table
Main_output_filtered_dt <- as.data.table(Main_output_filtered)
dMDT_proteomics_dt <- as.data.table(dMDT_proteomics)
MDT_proteomics_dt <- as.data.table(MDT_proteomics)

# Create logical vectors indicating matching IDs
dMDT_match <- sapply(Main_output_filtered_dt$Protein.Ids, function(ids) any(str_split_1(ids, ';') %in% dMDT_vector)) # Protein IDs that include dMDT
MDT_match <- sapply(Main_output_filtered_dt$Protein.Ids, function(ids) any(str_split_1(ids, ';') %in% MDT_vector)) # Protein IDs that include MDT

# Filter rows based on the logical vectors
dMDT_proteomics_dt <- Main_output_filtered_dt[dMDT_match]
MDT_proteomics_dt <- Main_output_filtered_dt[MDT_match]
```


```{r check_dMDTs_in_general}
library(dplyr)

# check rows dMDT is found together with other ENSP ids but excluding the MDTs 
library(purrr)
include_ids <- unique(Switches_Ovary_ensps$Protein.stable.ID)
exclude_ids <- unique(Switches_Ovary_ensps_mdts$Protein.stable.ID)

# Peptides that belong to dMDT and others but not to MDT ids
Main_output_filtered_dMDT_exc_MDT <- Main_output_filtered %>%
  dplyr::filter(
    str_detect(Protein.Ids, paste(include_ids, collapse = '|')) &
    !str_detect(Protein.Ids, paste(exclude_ids, collapse = '|'))
  )

## exlude Class I protein IDs to distinct Class I and Class II
Main_output_filtered_dMDT_exc_MDT_exc_ClassI <- Main_output_filtered_dMDT_exc_MDT %>%
  dplyr::filter(!Protein.Ids %in% Main_output_filtered_dMDT$Protein.Ids)


Main_output_filtered_dMDT_exc_MDT_exc_ClassI_sepRows <- Main_output_filtered_dMDT_exc_MDT_exc_ClassI %>% separate_rows(Protein.Ids, sep = ";") %>% dplyr::filter(Protein.Ids %in% Switches_Ovary_ensps$Protein.stable.ID) %>% dplyr::filter(!Protein.Ids %in% Main_output_filtered_dMDT$Protein.Ids)  # split the protein Ids and find cMDTs that are Class II - 164 Protein Ids found with Class II strategy  - Not Sample Specific!!
```


```{r focus_only_on_unique_peptides}
dMDT_proteomics_dt_unique_peptide <- dMDT_proteomics_dt[dMDT_proteomics_dt$Protein.Ids %in% Switches_Ovary_ensps$Protein.stable.ID, ] # 121 - dMDT specific peptides are counted.

MDT_proteomics_dt_unique_peptide <- MDT_proteomics_dt[MDT_proteomics_dt$Protein.Ids %in% Switches_Ovary_ensps_mdts$Protein.stable.ID, ] # unique MDT peptides - 311 MDT peptides are found in the samples. 

Swicthes_in_Dataset_gtex_tp_ov <- Switches_Ovary_ensps %>% dplyr::group_by(Transcript.stable.ID) %>% dplyr::summarize(n=n(), across()) %>% arrange(desc(n))
```

## length(unique(dMDT_proteomics_dt_unique_peptide$Protein.Ids)) 121 isoform has dMDT-specific peptides
## length(unique(MDT_proteomics_dt_unique_peptide$Protein.Ids)) 311 MDT isoforms has MDT-specific peptides in the samples

```{r CompareCommonSamples}
# prepare Datasets in RNA and DIA 
# sample wise comparison to see the percentage found across samples
## collect all Protein IDs in a row in a sample-specific way in RNA samples

#rna_each_sample_output <- Switches_Ovary_ensps[,c(3,9)] %>% dplyr::group_by(SampleID) %>% dplyr::mutate(Protein_IDS_in_Sample = paste(Protein.stable.ID, collapse = ',')) %>% dplyr::select(SampleID, Protein_IDS_in_Sample) %>% dplyr::distinct()

## Class I isoforms in the RNA Samples
rna_each_sample_output_3 <- Switches_Ovary_ensps[,c(3,9)] %>% dplyr::filter(Protein.stable.ID %in% Main_output_filtered_dMDT$Protein.Ids) %>% dplyr::group_by(SampleID) %>% dplyr::mutate(Protein_IDS_in_Sample = paste(Protein.stable.ID, collapse = ',')) %>% dplyr::select(SampleID, Protein_IDS_in_Sample) %>% dplyr::distinct()

## Class II isoforms in the RNA samples
rna_each_sample_output_4 <- Switches_Ovary_ensps[,c(3,9)] %>% dplyr::filter(Protein.stable.ID %in% Main_output_filtered_dMDT_exc_MDT_exc_ClassI_sepRows$Protein.Ids) %>% dplyr::group_by(SampleID) %>% dplyr::mutate(Protein_IDS_in_Sample = paste(Protein.stable.ID, collapse = ',')) %>% dplyr::select(SampleID, Protein_IDS_in_Sample) %>% dplyr::distinct()

## collect all Protein IDs in a row in a sample-specific way in DIA samples
dia_each_sample_output <- Main_output_filtered_dMDT[, c(2,4)] %>% dplyr::group_by(Run) %>% dplyr::mutate(Protein_IDS_in_Sample = paste(Protein.Ids, collapse = ',')) %>% dplyr::select(Run, Protein_IDS_in_Sample) %>% dplyr::distinct()

dia_each_sample_output_4 <- Main_output_filtered_dMDT_exc_MDT_exc_ClassI_sepRows[, c(2,4)] %>% dplyr::group_by(Run) %>% dplyr::mutate(Protein_IDS_in_Sample = paste(Protein.Ids, collapse = ',')) %>% dplyr::select(Run, Protein_IDS_in_Sample) %>% dplyr::distinct()

#rna_row_names <- do.call(rbind, strsplit(rna_each_sample_output$SampleID, '/'))[,2]

rna_row_names3 <- do.call(rbind, strsplit(rna_each_sample_output_3$SampleID, '/'))[,2]
rna_row_names4 <- do.call(rbind, strsplit(rna_each_sample_output_4$SampleID, '/'))[,2]
dia_row_names <- do.call(rbind, strsplit(dia_each_sample_output$Run, '-'))[,1]
dia_row_names4 <- do.call(rbind, strsplit(dia_each_sample_output_4$Run, '-'))[,1]


#rna_each_sample_output['Samples'] <- rna_row_names

rna_each_sample_output_3['Samples'] <- rna_row_names3
rna_each_sample_output_4['Samples'] <- rna_row_names4
dia_each_sample_output['Samples'] <- dia_row_names
dia_each_sample_output_4['Samples'] <- dia_row_names4

#rna_each_sample_output2 <- rna_each_sample_output %>% dplyr::select(Protein_IDS_in_Sample, Samples) %>% dplyr::group_by(Samples) %>% dplyr::mutate(Protein_IDS_in_Sample2 = paste(Protein_IDS_in_Sample, collapse = ',')) %>% dplyr::select(Samples, Protein_IDS_in_Sample2) %>% dplyr::distinct()

rna_each_sample_output3 <- rna_each_sample_output_3 %>% dplyr::select(Protein_IDS_in_Sample, Samples) %>% dplyr::group_by(Samples) %>% dplyr::mutate(Protein_IDS_in_Sample2 = paste(Protein_IDS_in_Sample, collapse = ',')) %>% dplyr::select(Samples, Protein_IDS_in_Sample2) %>% dplyr::distinct()

rna_each_sample_output4 <- rna_each_sample_output_4 %>% dplyr::select(Protein_IDS_in_Sample, Samples) %>% dplyr::group_by(Samples) %>% dplyr::mutate(Protein_IDS_in_Sample2 = paste(Protein_IDS_in_Sample, collapse = ',')) %>% dplyr::select(Samples, Protein_IDS_in_Sample2) %>% dplyr::distinct()
```

# Expression of dMDTs in the RNA seq data with or without peptide evidence.
```{r ExpMatrix}

ov_exp_matrix_1 <- readRDS('/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Ovary/TPMValues/RedundantTPMs/tuPro_tpm_NR_primary.RDS')

ov_exp_matrix_2 <- readRDS('/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Ovary/TPMValues/RedundantTPMs/tuPro_tpm_NR_metastatic.RDS')

ov_exp_matrix <- cbind(ov_exp_matrix_1, ov_exp_matrix_2[,3:ncol(ov_exp_matrix_2)])
```

```{r dMDTwPepEvidence}
#dMDT_with_peptide_evidence <- #unique(Swicthes_in_Dataset_gtex_tp_ov[Swicthes_in_Dataset_gtex_tp_ov$Protein.stable.ID %in% #unique(dMDT_proteomics_dt_unique_peptide$Protein.Ids), 'Transcript.stable.ID'])
#
#dMDT_without_peptide_evidence <- #unique(Swicthes_in_Dataset_gtex_tp_ov[!Swicthes_in_Dataset_gtex_tp_ov$Protein.stable.ID %in% #unique(dMDT_proteomics_dt_unique_peptide$Protein.Ids), 'Transcript.stable.ID'])

```

## Figure 5
```{r takeSampleSpecific}
coul2 <- c("violet","violetred4")
common_samples <- intersect(dia_each_sample_output$Samples, rna_each_sample_output3$Samples) #40 samples are common
all_sample <- data.frame(TranscriptLocation = as.numeric(), SampleID=as.character(), TPM=as.numeric(), Category=as.character())
for (each_sample in common_samples) {
  
    one_sample <- data.frame()
    
    sample_proteins_found <- dia_each_sample_output[dia_each_sample_output$Samples == each_sample, 'Protein_IDS_in_Sample']
    
    sample_proteins_found_vector <- strsplit(sample_proteins_found$Protein_IDS_in_Sample, ',')[[1]]
    
    sample_proteins_found_rna <- rna_each_sample_output3[rna_each_sample_output3$Samples == each_sample, 'Protein_IDS_in_Sample2']
    sample_proteins_found_rna_vector <- strsplit(sample_proteins_found_rna$Protein_IDS_in_Sample2, ',')[[1]]
    
    NonFoundMDP <- setdiff(sample_proteins_found_rna_vector, sample_proteins_found_vector)
    
    foundMDT <- intersect(sample_proteins_found_rna_vector, sample_proteins_found_vector)
    
    if(!is_empty(NonFoundMDP) & !is_empty(foundMDT) ) {
    
    dMDT_without_evi_dia <- ov_exp_matrix[ov_exp_matrix$ENST %in% ensg_enst_ensp_apris_v104[ensg_enst_ensp_apris_v104$Protein.stable.ID %in% NonFoundMDP, 'Transcript.stable.ID'], c('ENST',colnames(ov_exp_matrix[grepl(each_sample, colnames(ov_exp_matrix))]))]
    

    dMDT_with_evi_dia <- ov_exp_matrix[ov_exp_matrix$ENST %in% ensg_enst_ensp_apris_v104[ensg_enst_ensp_apris_v104$Protein.stable.ID %in% foundMDT, 'Transcript.stable.ID'], c('ENST',colnames(ov_exp_matrix[grepl(each_sample, colnames(ov_exp_matrix))]))]
    
    colnames(dMDT_with_evi_dia) <- c('ENST', 'TPM')
    dMDT_with_evi_dia$Category <- "PeptideEvidence"
    dMDT_with_evi_dia$SampleID <- each_sample
    
    
    colnames(dMDT_without_evi_dia) <- c('ENST', 'TPM')
    dMDT_without_evi_dia$Category <- "NoPeptideEvidence"
    dMDT_without_evi_dia$SampleID <- each_sample    
    
    one_sample <- rbind(dMDT_with_evi_dia, dMDT_without_evi_dia)
    all_sample <- rbind(all_sample, one_sample)
}}


ggplot(all_sample , aes(x=log10(TPM), color=Category)) +
  geom_density(lwd = 1.2, linetype = 1) +
  scale_color_manual(values=coul2) + theme_classic()
# length(all_sample$ENST) -- 118
```

wilcox.test(all_sample[all_sample$Category == 'NoPeptideEvidence', 'TPM'], all_sample[all_sample$Category == 'PeptideEvidence', 'TPM'])

## Class II expressions
```{r takeSampleSpecific_100dMDT}
coul2 <- c("green","darkgreen")
common_samples <- intersect(dia_each_sample_output_4$Samples, rna_each_sample_output4$Samples)

all_sample2 <- data.frame(TranscriptLocation = as.numeric(), SampleID=as.character(), TPM=as.numeric(), ENST = as.character(), Category=as.character())

for (each_sample in common_samples) {
  
    one_sample <- data.frame()
    
    sample_proteins_found <- dia_each_sample_output_4[dia_each_sample_output_4$Samples == each_sample, 'Protein_IDS_in_Sample']
    sample_proteins_found_vector <- strsplit(sample_proteins_found$Protein_IDS_in_Sample, ',')[[1]]
    
    sample_proteins_found_rna <- rna_each_sample_output4[rna_each_sample_output4$Samples == each_sample, 'Protein_IDS_in_Sample2']
    sample_proteins_found_rna_vector <- strsplit(sample_proteins_found_rna$Protein_IDS_in_Sample2, ',')[[1]]
    
    NonFoundMDP <- setdiff(sample_proteins_found_rna_vector, sample_proteins_found_vector)
    foundMDT <- intersect(sample_proteins_found_rna_vector, sample_proteins_found_vector)
    
    if(!is_empty(NonFoundMDP) & !is_empty(foundMDT) ) {
    
    dMDT_without_evi_dia <- ov_exp_matrix[ov_exp_matrix$ENST %in% ensg_enst_ensp_apris_v104[ensg_enst_ensp_apris_v104$Protein.stable.ID %in% NonFoundMDP, 'Transcript.stable.ID'], c('ENST',colnames(ov_exp_matrix[grepl(each_sample, colnames(ov_exp_matrix))]))]
    
    dMDT_with_evi_dia <- ov_exp_matrix[ov_exp_matrix$ENST %in% ensg_enst_ensp_apris_v104[ensg_enst_ensp_apris_v104$Protein.stable.ID %in% foundMDT, 'Transcript.stable.ID'], c('ENST',colnames(ov_exp_matrix[grepl(each_sample, colnames(ov_exp_matrix))]))]
    
    colnames(dMDT_with_evi_dia) <- c('ENST', 'TPM')
    dMDT_with_evi_dia$Category <- "PeptideEvidence"
    dMDT_with_evi_dia$SampleID <- each_sample
    
    
    colnames(dMDT_without_evi_dia) <- c('ENST', 'TPM')
    dMDT_without_evi_dia$Category <- "NoPeptideEvidence"
    dMDT_without_evi_dia$SampleID <- each_sample    
    
    
    one_sample <- rbind(dMDT_with_evi_dia, dMDT_without_evi_dia)

    all_sample2 <- rbind(all_sample2, one_sample)
}}


ggplot(all_sample2 , aes(x=log10(TPM), color=Category)) +
  geom_density(lwd = 1.2, linetype = 1) +
  scale_color_manual(values=coul2) + theme_classic()
```

wilcox.test(all_sample2[all_sample2$Category == 'NoPeptideEvidence', 'TPM'], all_sample2[all_sample2$Category == 'PeptideEvidence', 'TPM'])


```{r check_probability_findornot}
final_data <- data.frame()
for (each_sample in rna_each_sample_output3$Samples) {
  
  if (each_sample %in% dia_each_sample_output$Samples) {
    
    # take the proteins in a specific sample in RNA and DIA samples
    rna_sample <- rna_each_sample_output3[rna_each_sample_output3$Samples == each_sample, ]
    dia_sample <- dia_each_sample_output[dia_each_sample_output$Samples == each_sample, ]
    
    # count the number of proteins in the RNA sample
    count_dMDT_in_RNA <- length(unique(str_split_1(rna_sample$Protein_IDS_in_Sample2, ',')))
      
    # count the number of common proteins in the RNA and DIA sample 
    count_dMDT_in_protein <- length(intersect(
      unlist(strsplit(rna_sample$Protein_IDS_in_Sample2, ",")),
     unlist(strsplit(dia_sample$Protein_IDS_in_Sample, ","))
      ))
    
    # calculate the % of detected proteins in the protein sample
    percentage_occurance <- (count_dMDT_in_protein/count_dMDT_in_RNA)*100
    percentage_nonoccurance <- 100-percentage_occurance
    
    # list of proteins detected in both samples
    peptides <- unique(intersect(
      unlist(strsplit(rna_sample$Protein_IDS_in_Sample2, ",")),
     unlist(strsplit(dia_sample$Protein_IDS_in_Sample, ","))
      ))  
    
   if (!is_empty(peptides)) {
    data <- data.frame(sample = each_sample, percentage_occur = percentage_occurance, percentage_nonoccur = percentage_nonoccurance, peptides = peptides)     
    
   } else {
     
      data <- data.frame(sample = each_sample, percentage_occur = percentage_occurance, percentage_nonoccur = percentage_nonoccurance, peptides = 'None')    

   }
    
    final_data <- rbind(final_data, data)

}}
```
length(unique(final_data$sample))
[1] 40 common samples in RNA and DIA samples
# 40 proteins that is found in a sample-specific manner 

```{r stack_bar_ggplot}
final_data_allcMDTs <- data.frame()

for (each_sample in rna_each_sample_output4$Samples) {
  
  # Main_output_filtered_dMDT_exc_MDT_exlude_ClassI
  if (each_sample %in% dia_each_sample_output_4$Samples) {
    
    
    rna_sample <- rna_each_sample_output4[rna_each_sample_output4$Samples == each_sample, ]
    dia_sample <- dia_each_sample_output_4[dia_each_sample_output_4$Samples == each_sample, ]
    
    count_dMDT_in_RNA <- length(unique(str_split_1(rna_sample$Protein_IDS_in_Sample2, ',')))
      
    count_dMDT_in_protein <- length(unique(intersect(
      unlist(strsplit(rna_sample$Protein_IDS_in_Sample2, ",")),
     unlist(strsplit(dia_sample$Protein_IDS_in_Sample, ","))
      ))  )
    
    percentage_occurance <- (count_dMDT_in_protein/count_dMDT_in_RNA)*100
    percentage_nonoccurance <- 100-percentage_occurance
    
    peptides <- unique(intersect(
      unlist(strsplit(rna_sample$Protein_IDS_in_Sample2, ",")),
     unlist(strsplit(dia_sample$Protein_IDS_in_Sample, ","))
      ))   
    
   if (!is_empty(peptides)) {
    data_all_cmdt <- data.frame(sample = each_sample, percentage_occur = percentage_occurance, percentage_nonoccur = percentage_nonoccurance, peptides = peptides)  
    
   } else {
     
      data_all_cmdt <- data.frame(sample = each_sample, percentage_occur = percentage_occurance, percentage_nonoccur = percentage_nonoccurance, peptides = 'None')    

    
   }
    
    final_data_allcMDTs <- rbind(final_data_allcMDTs, data_all_cmdt)
}}

```

final_data_allcMDTs %>% dplyr::select(sample,percentage_occur) %>% dplyr::distinct() %>% dplyr::select(percentage_occur) %>% dplyr::summarise(meanPerc = mean(percentage_occur))
  meanPerc
1 45.71961

final_data %>% dplyr::select(sample,percentage_occur) %>% dplyr::distinct() %>% dplyr::select(percentage_occur) %>% dplyr::summarise(meanPerc = mean(percentage_occur))
  meanPerc
1  29.06461

length(unique(final_data_allcMDTs$peptides))
[1] 79
length(unique(final_data$peptides))
[1] 41

Increasing the percentage from 2% to 5% when we apply non-MDT strategy in 40 samples.

```{r splitNames}
Switches_Ovary_ensps$RunID <- do.call(rbind, strsplit(Switches_Ovary_ensps$SampleID, '/'))[,2]
Main_output_filtered_dMDT$RunID <- do.call(rbind, strsplit(Main_output_filtered_dMDT$Run, '-'))[,1]
```


```{r probability_in_proteomics}
## probability of those isoforms across Samples - here we do not count for sample specificty
common_samples <- intersect(dia_each_sample_output$Samples, rna_each_sample_output3$Samples) 

probabilit_proteomics_transcript <- final_data[,c(1,4)] %>% dplyr::distinct() %>% dplyr::group_by(peptides) %>% dplyr::summarize(n=n(), across()) %>% arrange(desc(n)) %>% dplyr::select(peptides, n) %>% dplyr::distinct()

rna_probabilit_proteomics_transcript <-  Switches_Ovary_ensps %>% dplyr::group_by(Transcript.stable.ID) %>% dplyr::filter(RunID %in% common_samples) %>% dplyr::summarize(n=n(), across()) %>% arrange(desc(n)) %>% dplyr::filter(Protein.stable.ID %in% final_data$peptides) %>% ungroup %>% dplyr::select(Protein.stable.ID, n) %>% dplyr::distinct()

probabilit_proteomics_transcript_ordered <- probabilit_proteomics_transcript[match(rna_probabilit_proteomics_transcript$Protein.stable.ID, probabilit_proteomics_transcript$peptides),]
```


```{r probability_in_proteomics_allcMDTs}
## probability of those isoforms  being seen in other samples 
probabilit_proteomics_79_transcript <- final_data_allcMDTs[,c(1,4)] %>% dplyr::distinct() %>% dplyr::group_by(peptides) %>% dplyr::summarize(n=n(), across()) %>% arrange(desc(n)) %>% dplyr::select(peptides, n) %>% dplyr::distinct()


#Main_output_filtered_dMDT_all_unnest <- Main_output_filtered_dMDT_exc_MDT_exc_ClassI_sepRows %>%
#  mutate(Protein.Ids = strsplit(as.character(Protein.Ids), ",")) %>%
#  unnest(Protein.Ids) 

Swicthes_in_Dataset_proteomics <- Swicthes_in_Dataset_gtex_tp_ov[Swicthes_in_Dataset_gtex_tp_ov$Protein.stable.ID %in% Main_output_filtered_dMDT_exc_MDT_exc_ClassI_sepRows$Protein.Ids, ]


rna_probabilit_proteomics_79_transcript <- Switches_Ovary_ensps %>% dplyr::group_by(Transcript.stable.ID) %>% dplyr::filter(RunID %in% common_samples) %>% dplyr::summarize(n=n(), across()) %>% arrange(desc(n)) %>% dplyr::filter(Protein.stable.ID %in% final_data_allcMDTs$peptides) %>% dplyr::ungroup() %>% dplyr::select(Protein.stable.ID, n) %>% dplyr::distinct()


#barplot(rna_probabilit_proteomics_116_transcript$n, col='#FDCDAC')
#barplot((rna_probabilit_proteomics_116_transcript$n/123*100), col='#FDCDAC')

probabilit_proteomics_79_transcript_ordered <- probabilit_proteomics_79_transcript[match(rna_probabilit_proteomics_79_transcript$Protein.stable.ID, probabilit_proteomics_79_transcript$peptides),]

#barplot(probabilit_proteomics_116_transcript_ordered$n, col='#B3E2CD')
#barplot((probabilit_proteomics_116_transcript_ordered$n/47*100), col='#B3E2CD')
```


```{r APRIS_annotations}
# APRIS annotation distribution of 121 dMDTs with non-MDT peptide evidence. 
probabilit_proteomics_transcript_ordered_apris <- merge(probabilit_proteomics_79_transcript_ordered, ensg_enst_ensp_apris_v104, by.x='peptides', by.y='Protein.stable.ID')

# APRIS annotation distribution of 41 dMDTs with non-MDT peptide evidence. 
probabilit_proteomics_transcript_ordered_apris_dMDTpep <- merge(probabilit_proteomics_transcript_ordered, ensg_enst_ensp_apris_v104, by.x='peptides', by.y='Protein.stable.ID')

```


```{r saveAPRIS}
apris_ovary <- ggplot(data=probabilit_proteomics_transcript_ordered_apris_dMDTpep, aes(APPRIS.annotation)) +
  geom_bar(color = "black", fill= "#C51B7D") +
  theme_classic()  + scale_fill_brewer(palette="Paired")

apris_ovary_2 <- ggplot(data=probabilit_proteomics_transcript_ordered_apris, aes(APPRIS.annotation, color=APPRIS.annotation)) +
  geom_bar(color = "black", fill='#7FBC41') +
  theme_classic()  + scale_fill_brewer(palette="Paired")

apris_ovary
apris_ovary_2
#  brewer.pal(n = 8, name = "PiYG")

#ggsave("/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Figures/apris_ovary_41E#NST.png", plot = apris_ovary, bg = "transparent", width = 5, height = 4)
#ggsave("/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Figures/apris_ovary_100#ENST.png", plot = apris_ovary_2, bg = "transparent", width = 5, height = 4)
```

length(unique(probabilit_proteomics_transcript_ordered_apris_dMDTpep$peptides))
[1] 41
probabilit_proteomics_transcript_ordered_apris_dMDTpep %>% dplyr::group_by(APPRIS.annotation) %>% dplyr::summarise(count = n())
24/41*100  58.53659
probabilit_proteomics_transcript_ordered_apris %>% dplyr::group_by(APPRIS.annotation) %>% dplyr::summarise(count = n())
20/79*100 25% class II

Most of the dMDTs that are found in the DIA are annotated as Principal 1, and some do not have any annotations. 
some notes: APRIS do not have Principal annotation for every gene. The ones without annotations might be the longest isoform or not having an annotation, or clinically relevant.


```{r checkPercentageOfIsoformsintheSampleSamples}
# find out % of samples that isoform found in the same sample that the isoform found as cMDT.
# Class I
probabilit_proteomics_transcript_ordered$Group <- 'Proteomics'
rna_probabilit_proteomics_transcript$Group <- 'RNA'

colnames(rna_probabilit_proteomics_transcript) <- c('peptides', 'n', 'Group')
rna_probabilit_proteomics_transcript$n2 <- rna_probabilit_proteomics_transcript$n/length(unique(common_samples))*100

probabilit_proteomics_transcript_ordered$n2 <- probabilit_proteomics_transcript_ordered$n/length(unique(final_data$sample))*100 

rna_probabilit_proteomics_79_transcript$Group <- 'RNA'
colnames(rna_probabilit_proteomics_79_transcript) <- c('peptides', 'n', 'Group')
rna_probabilit_proteomics_79_transcript$n2 <- rna_probabilit_proteomics_79_transcript$n/length(unique(common_samples))*100

probabilit_proteomics_79_transcript_ordered$Group <- 'Proteomics_Flex'
probabilit_proteomics_79_transcript_ordered$n2 <- probabilit_proteomics_79_transcript_ordered$n/length(unique(final_data_allcMDTs$sample))*100

probabilit_proteomics_79_transcript_selected <- probabilit_proteomics_79_transcript_ordered[probabilit_proteomics_79_transcript_ordered$peptides %in% probabilit_proteomics_transcript_ordered$peptides,]

probability_all <- rbind(probabilit_proteomics_transcript_ordered, probabilit_proteomics_79_transcript_selected, rna_probabilit_proteomics_transcript)
probability_all[probability_all$Group == 'RNA', ]$n2 <- (-probability_all[probability_all$Group == 'RNA', ]$n2) 

# Calculate order based on RNA values
probability_all_top20_peptides <- (head(probability_all %>% dplyr::filter(Group=='Proteomics') %>% dplyr::arrange(desc(n2)), 20) %>% dplyr::select(peptides))$peptides

probability_all_top20 <- probability_all[probability_all$peptides %in% probability_all_top20_peptides, ]

set.seed(43)
#ggplot(probability_all_top20, aes(x=peptides, y=n2, fill=Group)) +
#     geom_bar(data=subset(probability_all_top20, Group == "RNA"), aes(y=n2), stat="identity") +
#     geom_bar(data=subset(probability_all_top20, Group %in% c("Proteomics", "Proteomics_Flex")), 
#              aes(y=ifelse(Group == "Proteomics", n2, 0)), 
#              stat="identity", position=position_dodge(width=0.9)) +
#     geom_bar(data=subset(probability_all_top20, Group %in% c("Proteomics", "Proteomics_Flex")), 
#              aes(y=ifelse(Group == "Proteomics_Flex", n2, 0)), 
#              stat="identity", position=position_dodge(width=0.9)) +
#     coord_flip() +
#     scale_fill_manual(values=c('Proteomics'='darkviolet', 'Proteomics_Flex'='darkgreen', 'RNA'='#619CFF')) +
#     theme_classic()


probabilit_proteomics_79_transcript_uniq <- probabilit_proteomics_79_transcript_ordered[!probabilit_proteomics_79_transcript_ordered$peptides %in% probabilit_proteomics_transcript_ordered$peptides,]

probability_flex <- rbind(probabilit_proteomics_79_transcript_uniq,rna_probabilit_proteomics_79_transcript)

probability_flex[probability_flex$Group == 'RNA', ]$n2 <- (-probability_flex[probability_flex$Group == 'RNA', ]$n2) 


# Calculate order based on RNA values
probability_flex_top20_peptides <- (head(probability_flex %>% dplyr::filter(Group=='Proteomics_Flex') %>% dplyr::arrange(desc(n2)), 20) %>% dplyr::select(peptides))$peptides

probability_flex_top20 <- probability_flex[probability_flex$peptides %in% probability_flex_top20_peptides, ]

#ggplot(probability_flex_top20, aes(x=peptides, y=n2, fill=Group)) +
#     geom_bar(stat="identity") + coord_flip() + 
#  scale_fill_manual(values=c('Proteomics_Flex'='darkgreen', 'RNA'='#619CFF')) +
#  theme_classic()

# One sided bar graph - ordered based on RNA n2 which is at the background
ordered_peptides <- probability_all_top20 %>%
  filter(Group == "RNA") %>%
  arrange(desc(n2)) %>%
  pull(peptides)

probability_all_top20_ordered <- probability_all_top20 %>%
  mutate(peptides = factor(peptides, levels = ordered_peptides))

ggplot(probability_all_top20_ordered, aes(x=peptides, y=n2, fill=Group)) +
    geom_bar(data=subset(probability_all_top20_ordered, Group == "RNA"), aes(y=-n2), stat="identity", alpha=0.3) +
    geom_bar(data=subset(probability_all_top20_ordered, Group %in% c("Proteomics", "Proteomics_Flex")), 
             aes(y=ifelse(Group == "Proteomics", n2, 0)), 
             stat="identity", position=position_dodge(width=0.9)) +
    geom_bar(data=subset(probability_all_top20_ordered, Group %in% c("Proteomics", "Proteomics_Flex")), 
             aes(y=ifelse(Group == "Proteomics_Flex", n2, 0)), 
             stat="identity", position=position_dodge(width=0.9), alpha=0.3) +
    coord_flip() +
    scale_fill_manual(values=c('Proteomics'='darkviolet', 'Proteomics_Flex'='darkgreen', 'RNA'='#619CFF')) +
    theme_classic()
```

# Statistics -  if they are correlated or not. 
corr_occ <- cor.test(-probability_all_top20_ordered[probability_all_top20_ordered$Group == 'RNA','n2']$n2, probability_all_top20_ordered[probability_all_top20_ordered$Group == 'Proteomics','n2']$n2)
corr_occ$p.value
[1] 0.002999409
corr_occ$estimate
      cor 
0.6284781 

corr_occ_flex <- cor.test(-probability_all_top20_ordered[probability_all_top20_ordered$Group == 'RNA','n2']$n2, probability_all_top20_ordered[probability_all_top20_ordered$Group == 'Proteomics_Flex','n2']$n2)
corr_occ_flex$estimate
     cor 
0.648381 
corr_occ_flex$p.value
[1] 0.001987605


```{r savePlots}
#ov_Proteomics <- ggplot(all_sample , aes(x=log10(TPM), color=Category)) +
#  geom_density(lwd = 1.2, linetype = 1) +
#  scale_color_manual(values=coul2) + theme_classic()
#
#svg(filename = "/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Figures/ov_Pep_Evidence_Fig.svg", width = 8, #height = 6)
#print(ov_Proteomics)
#dev.off()
```


```{r checkExpression}
Switches_Ovary_ensps_top_proteins <- Switches_Ovary_ensps[Switches_Ovary_ensps$Protein.stable.ID %in% probabilit_proteomics_transcript_ordered$peptides,] %>% dplyr::group_by(Protein.stable.ID) %>% dplyr::summarise(meanTPMinCancer = mean(TPM1_cancer)) %>% dplyr::arrange(desc(meanTPMinCancer))

head(Switches_Ovary_ensps_top_proteins)
```

```{r graphHighlyExpressedOnes}
probability_flex_top20_exp <-  Switches_Ovary_ensps_top_proteins[1:20,]$Protein.stable.ID

probability_flex_top20_exp_ranking <- probability_all[probability_all$peptides %in% probability_flex_top20_exp, ]
# One sided bar graph - ordered based on RNA n2 which is at the background
probability_all_top20_ordered_exp <- probability_flex_top20_exp_ranking %>%
  mutate(peptides = factor(peptides, levels = rev(probability_flex_top20_exp)))

ggplot(probability_all_top20_ordered_exp, aes(x=peptides, y=n2, fill=Group)) +
    geom_bar(data=subset(probability_all_top20_ordered_exp, Group == "RNA"), aes(y=-n2), stat="identity", alpha=0.3) +
    geom_bar(data=subset(probability_all_top20_ordered_exp, Group %in% c("Proteomics", "Proteomics_Flex")), 
             aes(y=ifelse(Group == "Proteomics", n2, 0)), 
             stat="identity", position=position_dodge(width=0.9)) +
    geom_bar(data=subset(probability_all_top20_ordered_exp, Group %in% c("Proteomics", "Proteomics_Flex")), 
             aes(y=ifelse(Group == "Proteomics_Flex", n2, 0)), 
             stat="identity", position=position_dodge(width=0.9), alpha=0.3) +
    coord_flip() +
    scale_fill_manual(values=c('Proteomics'='darkviolet', 'Proteomics_Flex'='darkgreen', 'RNA'='#619CFF')) +
    theme_classic()
```
# Statistics -  if they are correlated or not. 
corr_exp <- cor.test(-probability_all_top20_ordered_exp[probability_all_top20_ordered_exp$Group == 'RNA','n2']$n2, probability_all_top20_ordered_exp[probability_all_top20_ordered_exp$Group == 'Proteomics','n2']$n2)
corr_exp$p.value
[1] 0.0007070069



## TCGA data check 
```{r read_cMDT_TCGA}
Switches_Ovary_tcga <- readRDS('/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Ovary/Switches/TCGA_GTEx_ov_Switch.RDS')
```

```{r checkTCGA}
# Class II peptide
Switches_Ovary_tcga[Switches_Ovary_tcga$dMDT %in% probabilit_proteomics_transcript_ordered_apris$Transcript.stable.ID, ]

## Class I peptide
unique(Switches_Ovary_tcga[Switches_Ovary_tcga$dMDT %in% probabilit_proteomics_transcript_ordered_apris_dMDTpep$Transcript.stable.ID, 'dMDT'])
```

Switches_Ovary_tcga[Switches_Ovary_tcga$dMDT %in% probabilit_proteomics_transcript_ordered_apris$Transcript.stable.ID, ] %>% dplyr::group_by(dMDT) %>% dplyr::summarise(count = n()) %>% dplyr::arrange(desc(count))
