---
title: "Manuscript 3: TuPro Thesis Analysis Pipeline - IsoNetwork and IsoDrug Network"
author: "TÃ¼lay Karakulak"
date: "2024-05-28"
output: 
  flexdashboard::flex_dashboard:
    storyboard: true
    social: menu
    source: embed
---

These codes represents the Figure 4, Supplementary Fig 6, and also other important figures and statistics regarding the Isoform and Drug networks. 

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r libraries}
library(patchwork)
library(dplyr)
library(ggplot2)
library(tidyverse)
library(knitr)
library(rmarkdown)
library(harmony)
library(Matrix)
library(cowplot)
library(RColorBrewer)
library(viridis)
library(ggpubr)
set.seed(12345)
```


### MDT Switch Analysis Across Cancer Cohorts
```{r read_switches}
dMDTs_Ovary_tcga <- readRDS('/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Ovary/Switches/CorrectedStat/TCGA_GTEx_ov_Switch.RDS')

dMDT_Ovary_tuPro_primary <- readRDS('/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Ovary/Switches/CorrectedStat/Switches_TuPro_Pri_GTEx.RDS')

dMDT_Ovary_tuPro_metastatis <- readRDS('/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Ovary/Switches/CorrectedStat/Switches_TuPro_Met_GTEx.RDS')

dMDT_Melanoma_tcga_primary <- readRDS('/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Melanoma/Switches/CorrectedStat/Switches_TCGA_Pri_GTEx.RDS')

dMDT_Melanoma_tcga_metastatis <- readRDS('/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Melanoma/Switches/CorrectedStat/Switches_TCGA_Met_GTEx.RDS')

dMDT_Melanoma_tupro <- readRDS('/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Melanoma/Switches/CorrectedStat/Switches_TuPro_GTEx.RDS')

dMDT_AML_tcga <- readRDS('/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/AML/Switches/CorrectedStat/Switches_TCGA_GTEx.RDS')

dMDT_AML_tuPro <- readRDS('/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/AML/Switches/CorrectedStat/Switches_TuPro_GTEx.RDS')
```


## Figure 4
### Isoform - Isoform Interaction Network 

```{r read_isoform_isoform_int_network}
# score 900
Iso_Iso_Network_v12 <- read.csv('/Users/tulaykarakulak/Documents/PhD/Projects/Iso_Iso_Network/interactionsInIsoforms_400_2_status_withTitin_withoutWARNING.tsv', sep='\t', header = TRUE)

head(Iso_Iso_Network_v12)

Iso_Iso_Network_v11 <- read.csv('/Users/tulaykarakulak/Downloads/interactionsInIsoforms_900_2.tsv', sep='\t', header = TRUE)

head(Iso_Iso_Network_v12)
head(Iso_Iso_Network_v11)
```

```{r read_string_PPI}
string_PPI <- read.csv('/Users/tulaykarakulak/Documents/PhD/Projects/Iso_Iso_Network/9606.protein.links.detailed.v12.0.txt', sep=' ', header = TRUE)

string_900PPI <- string_PPI %>% dplyr::filter(combined_score >= 900)
head(string_900PPI)
```

length(unique(string_900PPI$protein2)) # 12323
length(unique(string_900PPI$protein1)) # 12323
length(unique(string_400PPI$protein2)) # 19488

```{r visualize_Int_Network}
### STRING protein number  - Canonical
# 19699 -- sequence file ENSP - without 900 cutoff
string_canonical_protein_sequence <- 19699 
STRING_network_protein <- 19488 # with 400 score

### Total Transcript:Protein Number - Ensembl v99
Ensembl_transcript <- 111047
#111047      2

## Isoform-Isoform Interaction_Size
Number_of_Isoform_in_Network <- length(unique(Iso_Iso_Network_v12$ENST))

## No STRING interaction
No_STRING_interaction_gene_number <- (length(unique(Iso_Iso_Network_v12[Iso_Iso_Network_v12$STRINGintN == 0, 'ENSG'])) - 1) # removing one NA value

# No STRING interaction # ENST
No_STRING_interaction_enst_number <- (length(unique(Iso_Iso_Network_v12[Iso_Iso_Network_v12$STRINGintN == 0, 'ENST'])) - 1) # removing one NA value

## No domain-domain Interaction
No_3did_Info <- length(unique(Iso_Iso_Network_v12[Iso_Iso_Network_v12$ExistInts == '-' & Iso_Iso_Network_v12$MissInts == '-', 'ENST']))

## Proteins with Existing Interaction # No missed interaction
Only_existing_int <- length(unique(Iso_Iso_Network_v12[Iso_Iso_Network_v12$ExistInts != '-' & Iso_Iso_Network_v12$MissInts == '-', 'ENST']))


## Proteins with Missed Interaction -  No existing Interaction Left
Only_missed_int <- length(unique(Iso_Iso_Network_v12[Iso_Iso_Network_v12$ExistInts == '-' & Iso_Iso_Network_v12$MissInts != '-', 'ENST']))

Exist_missed_int <- length(unique(Iso_Iso_Network_v12[Iso_Iso_Network_v12$ExistInts != '-' & Iso_Iso_Network_v12$MissInts != '-', 'ENST']))
```

```{r names_numbers}
Names <- c('string_protein_sequence', 'STRING_network_protein', 'Ensembl_transcript', 'Number_of_Isoform_in_Network', 'No_STRING_interaction_enst_number', 'No_3did_Info', 'Only_existing_int', 'Only_missed_int', 'Exist_missed_int')
Numbers <- c(string_canonical_protein_sequence, STRING_network_protein, Ensembl_transcript, Number_of_Isoform_in_Network, No_STRING_interaction_enst_number, No_3did_Info, Only_existing_int, Only_missed_int, Exist_missed_int)
Names_Numbers <- data.frame(Names = Names, Numbers=Numbers)
```


dim(Iso_Iso_Network_v12)
[1] 99361    10. ## 99377    10 -- ## 900 combined score ## also 99377 with 400 combined score after adding Titin
dim(Iso_Iso_Network_v12[Iso_Iso_Network_v12$MissInts == '-', ])
[1] 82655    10 ## 89290    10 -- 900 combined score

dim(Iso_Iso_Network_v12[Iso_Iso_Network_v12$ExistInts == '-' & Iso_Iso_Network_v12$MissInts == '-', ])
[1] 69320    10 ## 81477    10  - 900 combined score
dim(Iso_Iso_Network_v12[Iso_Iso_Network_v12$ExistInts != '-' & Iso_Iso_Network_v12$MissInts == '-', ])
[1] 13335    10. ## 7813   10 -- 900 combined score 
dim(Iso_Iso_Network_v12[Iso_Iso_Network_v12$ExistInts == '-' & Iso_Iso_Network_v12$MissInts != '-', ])
[1] 16266    10 ## 9815   10 -- 900 combined score
dim(Iso_Iso_Network_v12[Iso_Iso_Network_v12$ExistInts != '-' & Iso_Iso_Network_v12$MissInts != '-', ])
[1] 456  10.   ## 272  10 -- 900 combined score

length(unique(Iso_Iso_Network_v12[Iso_Iso_Network_v12$ExistInts == '-' & Iso_Iso_Network_v12$MissInts != '-', 'ENSG']))
[1] 3993. ## 2396 900 combined score

length(unique(Iso_Iso_Network_v12$ENSG)) # 19700 genes in Iso Network # 99377 ENST ids. ## 19700 genes 
having NO domain domain int info: 14014 genes   ## [1] 16482 no domain info! length(unique(Iso_Iso_Network_v12[Iso_Iso_Network_v12$ExistInts == '-' & Iso_Iso_Network_v12$MissInts == '-', 'ENSG']))
having domain-domain info: 5686 genes # 3218 -- 900 combined score!

Genes having lost int: 3993 ENSG
Transcript having lost int: 16266 ENST

dim(Iso_Iso_Network_v12[Iso_Iso_Network_v12$ExistInts != '-' | Iso_Iso_Network_v12$MissInts != '-', ])
[1] 30057    10 # total transcript having some int 

## Figure Supplementary 6A - Network Number - Existing and Missing Pairs
```{r stack_bar}
library(viridis)
library(hrbrthemes)

Iso_Network_Statistics <- Names_Numbers[c(4,6:9),]
Struc_Info_3did <- Names_Numbers[Names_Numbers$Names == 'Number_of_Isoform_in_Network', 'Numbers'] - Names_Numbers[Names_Numbers$Names == 'No_3did_Info', 'Numbers']

Iso_Network_Statistics <- rbind(Iso_Network_Statistics, c('Struc_Info_3did', Struc_Info_3did))

category <- c('Network_Number', 'Structural_Info', 'Exit_Miss', 'Exit_Miss', 'Exit_Miss', 'Structural_Info')

Iso_Network_Statistics <- cbind(Iso_Network_Statistics, category)

Iso_Network_Statistics$Numbers <- as.numeric(Iso_Network_Statistics$Numbers)


category_order <- c("Network_Number", "Structural_Info", "Exit_Miss")

Iso_Network_Statistics$category <- factor(Iso_Network_Statistics$category, levels = category_order)

my_palette2 <- c("#000000",
"#94e1e3",
"#8d90ed",
"#57708f",
"#2c5689",
"#47adb7")
scale_color_manual(values = my_palette2)

plotStatistics <- ggplot(Iso_Network_Statistics, aes(x=category, y=Numbers, fill=Names)) + geom_col(position = "stack", stat='identity')  + coord_cartesian(ylim = c(0, max(Iso_Network_Statistics$Numbers) * 1.1)) + theme(axis.text.x = element_text(angle = 45, hjust = 1),  # Adjust the angle and horizontal justification of x-axis labels
        axis.text.y = element_text(size = 12),  # Adjust the font size of y-axis labels
        axis.title.x = element_text(size = 14),  # Adjust the font size of x-axis label
        axis.title.y = element_text(size = 14)) +
        scale_fill_manual(values = my_palette2) + theme_classic()

plotStatistics
# this figure can be Supplementary
#svg(filename = "/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Figures/Figure_IsoformNetwork.svg", width #= 7, height = 5)
#ggsave("/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Figures/Figure_IsoformNetwork.png", plot = xx, bg #= "transparent", width = 7, height = 5)
#print(xx)
#dev.off()
```

## Figure 5
### How many genes have cMDT  - how many of them have missing and kept interaction? 
```{r merge_all_cancer_ensgs}
dMDT_all_genes <- unique(c(dMDTs_Ovary_tcga$ENSG, dMDT_Ovary_tuPro_primary$ENSG, dMDT_Ovary_tuPro_metastatis$ENSG, dMDT_Melanoma_tcga_primary$ENSG, dMDT_Melanoma_tcga_metastatis$ENSG, dMDT_Melanoma_tupro$ENSG, dMDT_AML_tcga$ENSG, dMDT_AML_tuPro$ENSG))
```

length(dMDT_all_genes): 4151
total dMDT genes in three cohort: 4151

```{r kept_int_lost_int}
# merge_with the IsoNetwork to see the effect
dMDTs_Ovary_tcga$Cohort <- 'TCGA_Ovary'
dMDT_Ovary_tuPro_primary$Cohort <- 'TuPro_primary_Ovary'
dMDT_Ovary_tuPro_metastatis$Cohort <- 'TuPro_metastatic_Ovary'
dMDT_Melanoma_tcga_primary$Cohort <- 'TCGA_Primary_Melanoma'
dMDT_Melanoma_tcga_metastatis$Cohort <- 'TCGA_Metastatic_Melanoma'
dMDT_Melanoma_tupro$Cohort <- 'TuPro_Melanoma'
dMDT_AML_tcga$Cohort <- 'AML_TCGA'
dMDT_AML_tuPro$Cohort <- 'AML_TuPro'

dMDTs_all_cancers <- rbind(dMDTs_Ovary_tcga, dMDT_Ovary_tuPro_primary, dMDT_Ovary_tuPro_metastatis, dMDT_Melanoma_tcga_primary, dMDT_Melanoma_tcga_metastatis, dMDT_Melanoma_tupro, dMDT_AML_tcga, dMDT_AML_tuPro)

dMDTs_all_cancers_isoNetwork <- merge(dMDTs_all_cancers, Iso_Iso_Network_v12, by.x='dMDT', by.y='ENST')
```

```{r statisticsOncMDTIsoformNetwork}
domain_info <- dMDTs_all_cancers_isoNetwork %>% dplyr::filter(ExistIntN !=0 | MissIntN != 0) %>% dplyr::select(ENSG.x) %>% dplyr::distinct() %>% dplyr::count()
no_domain <- dMDTs_all_cancers_isoNetwork %>% dplyr::filter(ExistIntN ==0 & MissIntN == 0) %>% dplyr::select(ENSG.x) %>% dplyr::distinct() %>% dplyr::count()
# dMDTs_all_cancers_isoNetwork %>% dplyr::filter(ExistIntN !=0 | MissIntN != 0) %>% dplyr::select(dMDT) %>% dplyr::distinct() %>% dplyr::count() ENCODING 1652 dMDT

some_int_loss <- dMDTs_all_cancers_isoNetwork %>% dplyr::filter(ExistIntN !=0 & MissIntN != 0) %>% dplyr::select(dMDT) %>% dplyr::distinct() %>% dplyr::count()
kepts_all_int <- dMDTs_all_cancers_isoNetwork %>% dplyr::filter(ExistIntN != 0 & MissIntN == 0) %>% dplyr::select(dMDT) %>% dplyr::distinct() %>% dplyr::count()
missed_all_int <- dMDTs_all_cancers_isoNetwork %>% dplyr::filter(ExistIntN == 0 & MissIntN != 0) %>% dplyr::select(dMDT) %>% dplyr::distinct() %>% dplyr::count()



total_dMDT_ENSG <- dMDTs_all_cancers_isoNetwork %>% dplyr::select(dMDT) %>% dplyr::distinct() %>% dplyr::count()
total_ENSG <- Iso_Iso_Network_v12 %>% dplyr::select(ENSG) %>% dplyr::distinct() %>% dplyr::count()

print(paste("Some Int Loss:", some_int_loss$n))
print(paste("Kept All Int:", kepts_all_int$n))
print(paste("Missed All Int:", missed_all_int$n))
print(paste("No Domain Info:", no_domain$n))
print(paste("Available Domain Info:", domain_info$n))
print(paste("All Genes:", total_ENSG$n))
print(paste("All dMDT Genes:", total_dMDT_ENSG$n))
```

length of ENSG with some lost int: 1046

## Figure 5A
```{r graph}
dMDT_int_loss_profile_non_dMDT_genes <- (19700-3866)

dMDT_int_loss_profile <- data.frame(c(total_gene=total_ENSG$n, total_dMDT_gene = total_dMDT_ENSG$n, no_domain_dMDT=no_domain$n, some_int_lost = some_int_loss$n, kept_all_int = kepts_all_int$n, missed_all_int= missed_all_int$n, Non_dMDT_genes=dMDT_int_loss_profile_non_dMDT_genes, domain_info=domain_info$n))

dMDT_int_loss_profile$Types <- rownames(dMDT_int_loss_profile)
colnames(dMDT_int_loss_profile) <- c('Number', 'Types')

dMDT_int_loss_profile_geneNumber <- dMDT_int_loss_profile %>% filter(Types %in% c('Non_dMDT_genes', 'total_dMDT_gene'))

dMDT_domain_info <- dMDT_int_loss_profile %>% filter(Types %in% c('domain_info', 'no_domain_dMDT'))

dMDT_int_loss_profile_IntLoss <- dMDT_int_loss_profile %>% filter(Types %in% c('some_int_lost', 'kept_all_int', 'missed_all_int'))

hsize <- 1

dMDT_int_loss_profile_geneNumber <- dMDT_int_loss_profile_geneNumber %>% 
  mutate(x = hsize)

# gene number pie chart
plot1 <- ggplot(dMDT_int_loss_profile_geneNumber, aes(x = hsize, y = Number, fill = Types)) +
  geom_col(color = "black") +
  coord_polar(theta = "y") +
  scale_fill_manual(values = c("#74A9CF", "#0570B0")) +
  xlim(c(0.2, hsize + 0.5))  +
  theme(panel.background = element_rect(fill = "transparent", color = NA),
        plot.background = element_rect(fill = "transparent", color = NA),
        panel.grid = element_blank(),
        axis.title = element_blank(),
        axis.ticks = element_blank(),
        axis.text = element_blank())


# Domain Domain Info pie chart
dMDT_domain_info <- dMDT_domain_info %>% 
  mutate(x = hsize)

plot2 <- ggplot(dMDT_domain_info, aes(x = hsize, y = Number, fill = Types)) +
  geom_col(color = "black") +
  coord_polar(theta = "y") +
  scale_fill_manual(values = c("#3F3F91", "#6464D5")) +
  xlim(c(0.2, hsize + 0.5))  +
  theme(panel.background = element_rect(fill = "transparent", color = NA),
        plot.background = element_rect(fill = "transparent", color = NA),
        panel.grid = element_blank(),
        axis.title = element_blank(),
        axis.ticks = element_blank(),
        axis.text = element_blank())


# Int Lost Pie Chart
dMDT_int_loss_profile_IntLoss <- dMDT_int_loss_profile_IntLoss %>% 
  mutate(x = hsize)

plot3 <- ggplot(dMDT_int_loss_profile_IntLoss, aes(x = hsize, y = Number, fill = Types)) +
  geom_col(color = "black") + 
  coord_polar(theta = "y") +
  scale_fill_manual(values = c("#AFAFCA",
                               "#7678A0", "#8D8DBB")) + 
  xlim(c(0.2, hsize + 0.5))  +
  theme(panel.background = element_rect(fill = "transparent", color = NA),
        plot.background = element_rect(fill = "transparent", color = NA),
        panel.grid = element_blank(),
        axis.title = element_blank(),
        axis.ticks = element_blank(),
        axis.text = element_blank())


print(plot1)
print(plot2)
print(plot3)
```

### Integrate Isoform Interaction Network to dMDT Findings
```{r readExpression, warning=FALSE, message=FALSE}
ov_tcga_tpm_NR <- readRDS('/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Ovary/TPMValues/RedundantTPMs/tcga_tpm_NR.RDS')

ov_tuPro_tpm_NR_primary <- readRDS('/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Ovary/TPMValues/RedundantTPMs/tuPro_tpm_NR_primary.RDS')

ov_tuPro_tpm_NR_metastatic <- readRDS('/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Ovary/TPMValues/RedundantTPMs/tuPro_tpm_NR_metastatic.RDS')

mel_tcga_primary_tpm_NR <- readRDS('/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Melanoma/TPMValues/RedundantTPMs/tcga_primary_tpm_NR.RDS')

mel_tcga_metastatic_tpm_NR <- readRDS('/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Melanoma/TPMValues/RedundantTPMs/tcga_metastatic_tpm_NR.RDS')

mel_tuPro_tpm_NR <- readRDS('/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Melanoma/TPMValues/RedundantTPMs/tuPro_tpm_NR.RDS')

AML_tcga_tpm_NR <- readRDS('/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/AML/TPMValues/RedundantTPMs/tcga_tpm_NR.RDS')

AML_tuPro_tpm_NR <- readRDS('/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/AML/TPMValues/RedundantTPMs/tuPro_tpm_NR.RDS')
```


```{r buildNetwork, warning=FALSE, message=FALSE}
library(MDTToolset)
ENSG_ENST_ENSP_Pfam <- MDTToolset::ENSG_ENST_ENSP_Pfam

# Ovary TCGA Network
dMDTs_IsoNetwork_Ov_tcga <- merge(dMDTs_Ovary_tcga, Iso_Iso_Network_v12, by.x='dMDT', by.y='ENST')
dMDTs_IsoNetwork_int_Ov_tcga <- dMDTs_IsoNetwork_Ov_tcga[dMDTs_IsoNetwork_Ov_tcga$MissInts != '-' | dMDTs_IsoNetwork_Ov_tcga$ExistInts != '-',]

Switching_Events_Network_Ov_tcga <- MDTToolset::buildIsoNet(dMDT_network_info = dMDTs_IsoNetwork_int_Ov_tcga, ensembl_data = ENSG_ENST_ENSP_Pfam, kallisto_counts = ov_tcga_tpm_NR, iso_network = Iso_Iso_Network_v12)

## Ovary TuPro Network
dMDTs_IsoNetwork_Ov_tuPro <- merge(dMDT_Ovary_tuPro_primary, Iso_Iso_Network_v12, by.x='dMDT', by.y='ENST')
dMDTs_IsoNetwork_int_Ov_tuPro <- dMDTs_IsoNetwork_Ov_tuPro[dMDTs_IsoNetwork_Ov_tuPro$MissInts != '-' | dMDTs_IsoNetwork_Ov_tuPro$ExistInts != '-',]

Switching_Events_Network_Ov_tuPro_primary <- MDTToolset::buildIsoNet(dMDT_network_info = dMDTs_IsoNetwork_int_Ov_tuPro, ensembl_data = ENSG_ENST_ENSP_Pfam, kallisto_counts = ov_tuPro_tpm_NR_primary, iso_network = Iso_Iso_Network_v12)


## Ovary TuPro Metas Network

dMDTs_IsoNetwork_Ov_tuPro_met <- merge(dMDT_Ovary_tuPro_metastatis, Iso_Iso_Network_v12, by.x='dMDT', by.y='ENST')
dMDTs_IsoNetwork_int_Ov_tuPro_met <- dMDTs_IsoNetwork_Ov_tuPro_met[dMDTs_IsoNetwork_Ov_tuPro_met$MissInts != '-' | dMDTs_IsoNetwork_Ov_tuPro_met$ExistInts != '-',]

Switching_Events_Network_Ov_tuPro_met <- MDTToolset::buildIsoNet(dMDT_network_info = dMDTs_IsoNetwork_int_Ov_tuPro_met, ensembl_data = ENSG_ENST_ENSP_Pfam, kallisto_counts = ov_tuPro_tpm_NR_metastatic, iso_network = Iso_Iso_Network_v12)


## Melanoma TCGA Primary Network

dMDTs_IsoNetwork_Mel_tcga_pri <- merge(dMDT_Melanoma_tcga_primary, Iso_Iso_Network_v12, by.x='dMDT', by.y='ENST')

dMDTs_IsoNetwork_int_Mel_tcga_pri <- dMDTs_IsoNetwork_Mel_tcga_pri[dMDTs_IsoNetwork_Mel_tcga_pri$MissInts != '-' | dMDTs_IsoNetwork_Mel_tcga_pri$ExistInts != '-',]

Switching_Events_Network_Mel_tcga_Pri <- MDTToolset::buildIsoNet(dMDT_network_info = dMDTs_IsoNetwork_int_Mel_tcga_pri, ensembl_data = ENSG_ENST_ENSP_Pfam, kallisto_counts = mel_tcga_primary_tpm_NR, iso_network = Iso_Iso_Network_v12)


## Melanoma TCGA Met Network

dMDTs_IsoNetwork_Mel_tcga_met <- merge(dMDT_Melanoma_tcga_metastatis, Iso_Iso_Network_v12, by.x='dMDT', by.y='ENST')

dMDTs_IsoNetwork_int_Mel_tcga_met <- dMDTs_IsoNetwork_Mel_tcga_met[dMDTs_IsoNetwork_Mel_tcga_met$MissInts != '-' | dMDTs_IsoNetwork_Mel_tcga_met$ExistInts != '-',]

Switching_Events_Network_Mel_tcga_met  <- MDTToolset::buildIsoNet(dMDT_network_info = dMDTs_IsoNetwork_int_Mel_tcga_met, ensembl_data = ENSG_ENST_ENSP_Pfam, kallisto_counts = mel_tcga_metastatic_tpm_NR, iso_network = Iso_Iso_Network_v12)


## Melanoma TuPro Network
dMDTs_IsoNetwork_Mel_tuPro_met <- merge(dMDT_Melanoma_tupro, Iso_Iso_Network_v12, by.x='dMDT', by.y='ENST')

dMDTs_IsoNetwork_int_Mel_tuPro_met <- dMDTs_IsoNetwork_Mel_tuPro_met[dMDTs_IsoNetwork_Mel_tuPro_met$MissInts != '-' | dMDTs_IsoNetwork_Mel_tuPro_met$ExistInts != '-',]

Switching_Events_Network_Mel_tuPro_met  <- MDTToolset::buildIsoNet(dMDT_network_info = dMDTs_IsoNetwork_int_Mel_tuPro_met, ensembl_data = ENSG_ENST_ENSP_Pfam, kallisto_counts = mel_tuPro_tpm_NR, iso_network = Iso_Iso_Network_v12)



## AML TCGA Network
dMDTs_IsoNetwork_AML_tcga <- merge(dMDT_AML_tcga, Iso_Iso_Network_v12, by.x='dMDT', by.y='ENST')

dMDTs_IsoNetwork_int_AML_TCGA <- dMDTs_IsoNetwork_AML_tcga[dMDTs_IsoNetwork_AML_tcga$MissInts != '-' | dMDTs_IsoNetwork_AML_tcga$ExistInts != '-',]

Switching_Events_Network_AML_TCGA  <- MDTToolset::buildIsoNet(dMDT_network_info = dMDTs_IsoNetwork_int_AML_TCGA, ensembl_data = ENSG_ENST_ENSP_Pfam, kallisto_counts = AML_tcga_tpm_NR, iso_network = Iso_Iso_Network_v12)

```


```{r AMLcont}
## AML tuPro Network
dMDTs_IsoNetwork_AML_tupro <- merge(dMDT_AML_tuPro, Iso_Iso_Network_v12, by.x='dMDT', by.y='ENST')
dMDTs_IsoNetwork_int_AML_TuPro <- dMDTs_IsoNetwork_AML_tupro[dMDTs_IsoNetwork_AML_tupro$MissInts != '-' | dMDTs_IsoNetwork_AML_tupro$ExistInts != '-',]
Switching_Events_Network_AML_TuPro  <- MDTToolset::buildIsoNet(dMDT_network_info = dMDTs_IsoNetwork_int_AML_TuPro, ensembl_data = ENSG_ENST_ENSP_Pfam, kallisto_counts = AML_tuPro_tpm_NR, iso_network = Iso_Iso_Network_v12)
```


#```{r saveResultsNetwork}
#saveRDS(Switching_Events_Network_AML_TuPro, #'/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/AML/Switching_Events_NetworkChange_AML_TuPro.RDS')
#saveRDS(Switching_Events_Network_AML_TCGA, #'/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/AML/Switching_Events_NetworkChange_AML_TCGA.RDS')
#
#saveRDS(Switching_Events_Network_Mel_tuPro_met, #'/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Melanoma/Switching_Events_NetworkChange_Mel_tuPro_met.RD#S')
#
#saveRDS(Switching_Events_Network_Mel_tcga_met, #'/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Melanoma/Switching_Events_NetworkChange_Mel_tcga_met.RDS#')
#
#saveRDS(Switching_Events_Network_Mel_tcga_Pri, #'/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Melanoma/Switching_Events_NetworkChange_Mel_tcga_Pri.RDS#')
#
#saveRDS(Switching_Events_Network_Ov_tuPro_met, #'/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Ovary/Switching_Events_NetworkChange_Ov_tuPro_met.RDS')
#
#saveRDS(Switching_Events_Network_Ov_tuPro_primary, #'/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Ovary/Switching_Events_NetworkChange_Ov_tuPro_primary.RD#S')
#
#saveRDS(Switching_Events_Network_Ov_tcga, #'/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Ovary/Switching_Events_NetworkChange_Ov_tcga.RDS')
#```


### Isolate Int Losses
```{r IsolateIntLoss}
# Function to isolate only ENSP IDs
isolate_ensp_ids <- function(x) {
  # Split the string by commas
  parts <- unlist(strsplit(x, ","))
  # Filter out only parts that start with "ENSP"
  ensp_ids <- parts[grep("^ENSP", parts)]
  # Collapse the result back to a single string, if needed
  paste(ensp_ids, collapse = ",")
}

# Apply the function to the column
Switching_Events_Network_Ov_tcga$ensps_missed <- sapply(Switching_Events_Network_Ov_tcga$ensp_expressed_missed, isolate_ensp_ids)
Switching_Events_Network_Ov_tcga$ensps_kept <- sapply(Switching_Events_Network_Ov_tcga$ensp_expressed_kept, isolate_ensp_ids)


# Apply the function to the column
Switching_Events_Network_Ov_tuPro_primary$ensps_missed <- sapply(Switching_Events_Network_Ov_tuPro_primary$ensp_expressed_missed, isolate_ensp_ids)
Switching_Events_Network_Ov_tuPro_primary$ensps_kept <- sapply(Switching_Events_Network_Ov_tuPro_primary$ensp_expressed_kept, isolate_ensp_ids)


# Apply the function to the column
Switching_Events_Network_Ov_tuPro_met$ensps_missed <- sapply(Switching_Events_Network_Ov_tuPro_met$ensp_expressed_missed, isolate_ensp_ids)
Switching_Events_Network_Ov_tuPro_met$ensps_kept <- sapply(Switching_Events_Network_Ov_tuPro_met$ensp_expressed_kept, isolate_ensp_ids)


# Apply the function to the column
Switching_Events_Network_Mel_tcga_Pri$ensps_missed <- sapply(Switching_Events_Network_Mel_tcga_Pri$ensp_expressed_missed, isolate_ensp_ids)
Switching_Events_Network_Mel_tcga_Pri$ensps_kept <- sapply(Switching_Events_Network_Mel_tcga_Pri$ensp_expressed_kept, isolate_ensp_ids)


# Apply the function to the column
Switching_Events_Network_Mel_tcga_met$ensps_missed <- sapply(Switching_Events_Network_Mel_tcga_met$ensp_expressed_missed, isolate_ensp_ids)
Switching_Events_Network_Mel_tcga_met$ensps_kept <- sapply(Switching_Events_Network_Mel_tcga_met$ensp_expressed_kept, isolate_ensp_ids)

# Apply the function to the column
Switching_Events_Network_Mel_tuPro_met$ensps_missed <- sapply(Switching_Events_Network_Mel_tuPro_met$ensp_expressed_missed, isolate_ensp_ids)
Switching_Events_Network_Mel_tuPro_met$ensps_kept <- sapply(Switching_Events_Network_Mel_tuPro_met$ensp_expressed_kept, isolate_ensp_ids)

# Apply the function to the column
Switching_Events_Network_AML_TCGA$ensps_missed <- sapply(Switching_Events_Network_AML_TCGA$ensp_expressed_missed, isolate_ensp_ids)
Switching_Events_Network_AML_TCGA$ensps_kept <- sapply(Switching_Events_Network_AML_TCGA$ensp_expressed_kept, isolate_ensp_ids)

# Apply the function to the column
Switching_Events_Network_AML_TuPro$ensps_missed <- sapply(Switching_Events_Network_AML_TuPro$ensp_expressed_missed, isolate_ensp_ids)
Switching_Events_Network_AML_TuPro$ensps_kept <- sapply(Switching_Events_Network_AML_TuPro$ensp_expressed_kept, isolate_ensp_ids)

```



```{r SumAllResults}
Switching_Events_Network_AML_TuPro$Cohort <- 'TuPro_AML'
Switching_Events_Network_AML_TCGA$Cohort <- 'TCGA_AML'
Switching_Events_Network_Mel_tuPro_met$Cohort <- 'TuPro_Mel'
Switching_Events_Network_Mel_tcga_met$Cohort <- 'TCGA_Metastatic_Mel'
Switching_Events_Network_Mel_tcga_Pri$Cohort <- 'TCGA_Primary_Mel'
Switching_Events_Network_Ov_tuPro_met$Cohort <- 'TuPro_Metastatic_Ovary'
Switching_Events_Network_Ov_tuPro_primary$Cohort <- 'TuPro_Primary_Ovary'
Switching_Events_Network_Ov_tcga$Cohort  <- 'TCGA_Ovary'

Switches_Events_All <- rbind(Switching_Events_Network_AML_TuPro, Switching_Events_Network_AML_TCGA, Switching_Events_Network_Mel_tuPro_met, Switching_Events_Network_Mel_tcga_met, Switching_Events_Network_Mel_tcga_Pri, Switching_Events_Network_Ov_tuPro_met, Switching_Events_Network_Ov_tuPro_primary, Switching_Events_Network_Ov_tcga)

#saveRDS(Switches_Events_All, '/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Switches_Events_NetworkDistruption_All.RDS')
```


```{r sepRows}
length(unique(Switches_Events_All[Switches_Events_All$ensps_missed != '' & Switches_Events_All$ensps_kept == '', 'ENSG.x']))  # number of ENSGs with missed interactions
length(unique(Switches_Events_All[Switches_Events_All$ensps_missed == '' & Switches_Events_All$ensps_kept != '', 'ENSG.x']))  # number of ENSGs with kepts interactions 
length(unique(Switches_Events_All[Switches_Events_All$ensps_missed == '' & Switches_Events_All$ensps_kept == '', 'ENSG.x']))  # number of ENSGs where the int partners are not expressed or No missed interaction 
length(unique(Switches_Events_All[Switches_Events_All$ensps_missed != '' & Switches_Events_All$ensps_kept != '', 'ENSG.x']))  # number of ENSGs where there are some int partner lost their int while some kept 
```


## Isoform Drug Interaction Integration
```{r read_drugIso}
isoform_drug_database <- readRDS( file='/Users/tulaykarakulak/Documents/PhD/Projects/Drug_Isoform/DrugIsoformInt/isoform_drug_database_3.RDS')
```


```{r statistics}
length(unique(isoform_drug_database$gene))
length(unique(isoform_drug_database$reference_enst))
length(unique(isoform_drug_database$isoform))
length(unique(isoform_drug_database$drug_name))
```
```{r codingseq}
ensg_enst_ensp_pdb_pfams <- read.csv('/Users/tulaykarakulak/Documents/PhD/Projects/Drug_Isoform/mart_export.txt', sep='\t', header=TRUE)
head(ensg_enst_ensp_pdb_pfams)

protein_coding_ensts <- ensg_enst_ensp_pdb_pfams[ensg_enst_ensp_pdb_pfams$Transcript.type == 'protein_coding', 2]

isoform_drug_database_protein_coding <- isoform_drug_database[isoform_drug_database$isoform %in% protein_coding_ensts,]

length(unique(isoform_drug_database_protein_coding$gene))
length(unique(isoform_drug_database_protein_coding$reference_enst))
length(unique(isoform_drug_database_protein_coding$isoform))
length(unique(isoform_drug_database_protein_coding$drug_name))
```

```{r statisticsyy}
# bargraph - number of ENSG, ENST, drug, reference ENST
gene_number <- length(unique(isoform_drug_database_protein_coding$gene))
reference_enst <- length(unique(isoform_drug_database_protein_coding$reference_enst))
isoform_number <- length(unique(isoform_drug_database_protein_coding$isoform))
drug_number <- length(unique(isoform_drug_database_protein_coding$drug_name))

ENSG_list_all_all_disrupted <- as.vector(dMDTs_all_cancers_isoNetwork %>% dplyr::filter(ExistIntN !=0 | MissIntN != 0) %>% dplyr::select(ENSG.x) %>% dplyr::distinct())
dMDT_gene_targeted_by_drug <- isoform_drug_database_protein_coding[isoform_drug_database_protein_coding$gene %in% ENSG_list_all_all_disrupted$ENSG.x, ]
dMDT_gene_targeted_by_drug2 <- isoform_drug_database_protein_coding[isoform_drug_database_protein_coding$isoform %in%  dMDTs_all_cancers_isoNetwork$dMDT, ]
```



```{r check_howmany}
#total_length_of_dDMT_gene <- length(ENSG_list_all_all_disrupted$ENSG.x)
#Non_targeted_length_of_drugdMDT_gene <- (total_length_of_dDMT_gene - length(unique(dMDTs_all_cancers$gene)))
#total_length_of_drugdMDT_gene <- length(unique(dMDTs_all_cancers$gene))

total_length_of_dDMT_gene <- length(unique(dMDTs_all_cancers_isoNetwork$ENSG.x)) # but in the Iso Network it is less
Number_of_Gene_w_Structure <- length(unique(dMDTs_all_cancers_isoNetwork[dMDTs_all_cancers_isoNetwork$ENSG.x %in% isoform_drug_database_protein_coding$gene,'ENSG.x']))
Number_of_Gene_wo_Structure <- (total_length_of_dDMT_gene - Number_of_Gene_w_Structure)

dMDT_int_loss_drug_profile <- data.frame(c(GenewStructure=Number_of_Gene_w_Structure, total_length_of_dMDT_gene = (total_length_of_dDMT_gene-Number_of_Gene_w_Structure)))

dMDT_int_loss_drug_profile$Types <- rownames(dMDT_int_loss_drug_profile)
colnames(dMDT_int_loss_drug_profile) <- c('Number', 'Types')

# How many genes have interactions with Drugs
dMDT_int_loss_drug_profile <- dMDT_int_loss_drug_profile %>% 
  mutate(x = hsize)

plot4 <- ggplot(dMDT_int_loss_drug_profile, aes(x = hsize, y = Number, fill = Types)) +
  geom_col(color = "black") + 
  coord_polar(theta = "y") +
  scale_fill_manual(values = c("#80d5b0",
"#c3e0d4")) + 
  xlim(c(0.2, hsize + 0.5))  +
  theme(panel.background = element_rect(fill = "transparent", color = NA),
        plot.background = element_rect(fill = "transparent", color = NA),
        panel.grid = element_blank(),
        axis.title = element_blank(),
        axis.ticks = element_blank(),
        axis.text = element_blank())

plot4
```

#```{r saveplots2}
#svg(filename = "/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Figures/Drug_Isoform_StructureAvailibity.#svg", width = 7, height = 7)
#print(plot4)
#dev.off()
#```

```{r LostInt}
# How many genes have Lost Interactions
All_kept_int <- dMDT_gene_targeted_by_drug2 %>% dplyr::select(isoform, interaction) %>% dplyr::filter(interaction=='YES') %>% dplyr::distinct() %>% dplyr::count()

total_missed_int <- dMDT_gene_targeted_by_drug2 %>% dplyr::select(isoform, interaction) %>% dplyr::filter(interaction=='NO') %>%  dplyr::distinct() %>% dplyr::count()

dMDT_int_missed <- data.frame(c(total_kept_int=All_kept_int$n, total_missed_int = total_missed_int$n))

dMDT_int_missed$Types <- rownames(dMDT_int_missed)
colnames(dMDT_int_missed) <- c('Number', 'Types')
  
dMDT_int_missed <- dMDT_int_missed %>% 
  mutate(x = hsize)

plot5 <- ggplot(dMDT_int_missed, aes(x = hsize, y = Number, fill = Types)) +
  geom_col(color = "black") + 
  coord_polar(theta = "y") +
  scale_fill_manual(values = c("#cfdbd6",
"#bddcdc")) + 
  xlim(c(0.2, hsize + 0.5))  +
  theme(panel.background = element_rect(fill = "transparent", color = NA),
        plot.background = element_rect(fill = "transparent", color = NA),
        panel.grid = element_blank(),
        axis.title = element_blank(),
        axis.ticks = element_blank(),
        axis.text = element_blank())

print(plot5)
```



#```{r saveplots5}
#svg(filename = "/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Figures/Drug_Isoform_YESNO.svg", width = #7, height = 7)
#print(plot5)
#dev.off()
#```


dMDT_gene_targeted_by_drug2 %>% dplyr::select(isoform, drug_name) %>% dplyr::distinct() %>% dplyr::count() 1047 isoform:drug interaction for dMDTs


### bargraph - not included in the manuscript
```{r bargraphForDrugIsoformDB}
gene_number <- length(unique(isoform_drug_database_protein_coding$gene))
reference_enst <- length(unique(isoform_drug_database_protein_coding$reference_enst))
isoform_number <- length(unique(isoform_drug_database_protein_coding$isoform))
drug_number <- length(unique(isoform_drug_database_protein_coding$drug_name))
pdb_number <- length(unique(isoform_drug_database_protein_coding$pdb))

IsoDrug_Network_Statistics <- as.data.frame(t(data.frame(gene_number=gene_number, reference_enst=reference_enst, isoform_number=isoform_number, drug_number=drug_number, pdb_number=pdb_number)))
colnames(IsoDrug_Network_Statistics) <- c('Number')
IsoDrug_Network_Statistics$Category <- rownames(IsoDrug_Network_Statistics)

plot6 <- ggplot(IsoDrug_Network_Statistics, aes(x=Category, y=Number, fill=Category)) + geom_col(position = "stack", stat='identity') + theme_classic() 
plot6
#svg(filename = "/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Figures/NumberOFIsoformsDrugsGenesPdbS.sv#g", width = 7, height = 7)
#print(plot6)
#dev.off()



gene_number_dMDT <- length(unique(dMDT_gene_targeted_by_drug2$gene))
reference_enst_dMDT <- length(unique(dMDT_gene_targeted_by_drug2$reference_enst))
isoform_number_dMDT <- length(unique(dMDT_gene_targeted_by_drug2$isoform))
drug_number_dMDT <- length(unique(dMDT_gene_targeted_by_drug2$drug_name))
pdb_number_dMDT <- length(unique(dMDT_gene_targeted_by_drug2$pdb))

IsoDrug_Network_Statistics_dMDT <- as.data.frame(t(data.frame(gene_number=gene_number_dMDT, reference_enst=reference_enst_dMDT, isoform_number=isoform_number_dMDT, drug_number=drug_number_dMDT, pdb_number=pdb_number_dMDT)))
colnames(IsoDrug_Network_Statistics_dMDT) <- c('Number')
IsoDrug_Network_Statistics_dMDT$Category <- rownames(IsoDrug_Network_Statistics_dMDT)

plot7 <- ggplot(IsoDrug_Network_Statistics_dMDT, aes(x=Category, y=Number, fill=Category)) + geom_col(position = "stack", stat='identity') + theme_classic() 
plot7
#svg(filename = "/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/Figures/NumberOFIsoformsDrugsGenesPdb_dMD#T.svg", width = 7, height = 7)
#print(plot7)
#dev.off()
```

## Tyrosine Kinase Inhibitors
```{r TKIlist}
TKI <- read.table('/Users/tulaykarakulak/Documents/PhD/Projects/Drug_Isoform/DrugBank/TKI_Drugs_28052024.txt', header = TRUE, sep='\t')
head(TKI)
TKI$DrugName <- tolower(TKI$DrugName)
```


```{r readDrugNames}
drug_het_ids <- read.csv('/Users/tulaykarakulak/Documents/PhD/Projects/Drug_Isoform/DrugIsoformInt/Drug_Het_Ids.tsv', sep=',')
head(drug_het_ids)
```

```{r checkThose}
TKI_drug <- drug_het_ids[drug_het_ids$genericName %in% TKI$DrugName,]
dMDT_TKI_drug <- dMDT_gene_targeted_by_drug2[dMDT_gene_targeted_by_drug2$drug_name %in% TKI_drug$hetName, ]
head(dMDT_TKI_drug)
```
unique(unlist(dMDT_TKI_drug$drug_name))
TKI: "AQ4" "FMM" "IRE" "1E8" "BAX" "STI" "1N1" "032" "NIL" "AXI" "B49"
drug_het_ids[drug_het_ids$hetName %in% unique(unlist(dMDT_TKI_drug$drug_name)),'genericName']

TKI[TKI$DrugName %in% drug_het_ids[drug_het_ids$hetName %in% unique(unlist(dMDT_TKI_drug$drug_name)),'genericName'],] # description of drugs


```{r CheckTKENSTs}
dMDTs_all_cancers_TKI_Drug <- dMDTs_all_cancers %>% dplyr::group_by(Cohort, dMDT) %>%  dplyr::summarise(count = n()) %>% dplyr::filter(dMDT %in% dMDT_TKI_drug$isoform) %>% dplyr::arrange(desc(count))
dMDTs_all_cancers_TKI_Drug_lost_int <- dMDTs_all_cancers_TKI_Drug[dMDTs_all_cancers_TKI_Drug$dMDT %in% dMDT_TKI_drug[dMDT_TKI_drug$interaction == 'NO', 'isoform'],]
head(dMDTs_all_cancers_TKI_Drug_lost_int)
```

## Supplementary Figure 8: Cancer Census Genes Calculation
```{r cancercensusgenes}
CCG <- read.csv('/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/CancerGeneCensus/Cosmic_CancerGeneCensus_Tsv_v98_GRCh38/Cosmic_CancerGeneCensus_v98_GRCh38.tsv', sep='\t', header = TRUE)

head(CCG)
```

```{r  proteinidCCG}
string_PPI$protein1 <- gsub("^9606.", "", string_PPI$protein1)
string_PPI$protein2 <- gsub("^9606.", "", string_PPI$protein2)

ENSG_Protein_Match_CCG <- merge(CCG, ensg_enst_ensp_pdb_pfams[, c(1,2,3,10)], by.x='GENE_SYMBOL', 'Gene.name')
ENSG_Protein_Match_CCG_canonical <- unique(ENSG_Protein_Match_CCG[ENSG_Protein_Match_CCG$Protein.stable.ID %in% string_PPI$protein1, 'Protein.stable.ID'])
```
unique(dMDTs_all_cancers[dMDTs_all_cancers$ENSG %in% ENSG_Protein_Match_CCG$Gene.stable.ID,  'ENSG'])


length(unique(Switches_Events_All[Switches_Events_All$ENSG.x %in% ENSG_Protein_Match_CCG$Gene.stable.ID, 'ENSG.x']))

```{r  proteiniddMDT}
# disrupted
string_400PPI <- string_PPI %>% dplyr::filter(combined_score >= 400)

string_400PPI$protein1 <- gsub("^9606.", "", string_400PPI$protein1)
string_400PPI$protein2 <- gsub("^9606.", "", string_400PPI$protein2)


ENSG_list_some_disrupted <- as.vector(Switches_Events_All %>% dplyr::filter(ensps_missed != '' & ensps_kept == '') %>% dplyr::select(ENSG.x) %>% dplyr::distinct())

ENSG_Protein_Match_dMDT_distruptedPPI <- ensg_enst_ensp_pdb_pfams[ensg_enst_ensp_pdb_pfams$Gene.stable.ID %in% ENSG_list_some_disrupted$ENSG.x, ]

ENSG_Protein_Match_dMDT_distruptedPPI_canonical <- unique(ENSG_Protein_Match_dMDT_distruptedPPI[ENSG_Protein_Match_dMDT_distruptedPPI$Protein.stable.ID %in% string_400PPI$protein1, 'Protein.stable.ID'])


# Non Disrupted
ENSG_list_all_all_NonDisrupted <- unique(dMDTs_all_cancers[!dMDTs_all_cancers$ENSG %in% ENSG_list_some_disrupted$ENSG.x, 'ENSG'])

ENSG_Protein_Match_dMDT_NondistruptedPPI <- ensg_enst_ensp_pdb_pfams[ensg_enst_ensp_pdb_pfams$Gene.stable.ID %in% ENSG_list_all_all_NonDisrupted, ]

ENSG_Protein_Match_dMDT_NondistruptedPPI_canonical <- unique(ENSG_Protein_Match_dMDT_NondistruptedPPI[ENSG_Protein_Match_dMDT_NondistruptedPPI$Protein.stable.ID %in% string_400PPI$protein1, 'Protein.stable.ID'])

```


```{r calculateDistances}
library(igraph)
#string_400PPI <- string_PPI %>% dplyr::filter(combined_score >= 400)
#
#string_400PPI$protein1 <- gsub("^9606.", "", string_400PPI$protein1)
#string_400PPI$protein2 <- gsub("^9606.", "", string_400PPI$protein2)

graph <- graph.data.frame(string_400PPI, directed = FALSE)

cancer_census <- intersect(V(graph)$name, ENSG_Protein_Match_CCG_canonical)

dist_matrix_disrupted <- distances(graph, v = ENSG_Protein_Match_dMDT_distruptedPPI_canonical, to = ENSG_Protein_Match_CCG_canonical) 

# select only ones found in STRING network
ENSG_Protein_Match_dMDT_NondistruptedPPI_canonical_stringavailable <- ENSG_Protein_Match_dMDT_NondistruptedPPI_canonical[ENSG_Protein_Match_dMDT_NondistruptedPPI_canonical %in% string_400PPI$protein1] 

dist_matrix_Nondisrupted <- distances(graph, v = ENSG_Protein_Match_dMDT_NondistruptedPPI_canonical_stringavailable, to = ENSG_Protein_Match_CCG_canonical)


## Random DATA
# Create a random set of proteins
set.seed(123)  # For reproducibility
random_proteins <- sample(V(graph)$name, 500)

# Calculate shortest paths
random_paths <- distances(graph, v = random_proteins, to = ENSG_Protein_Match_CCG_canonical)

# Find the maximum distance
#max_distance <- max(random_paths, na.rm = TRUE)
#print(max_distance)
```

How many cancer census genes in our dMDT? 
length(unique(intersect(ENSG_Protein_Match_dMDT_distruptedPPI_canonical, ENSG_Protein_Match_CCG_canonical)))
[1] 50

```{r df}
distances_df_Disrupted <- as.data.frame(dist_matrix_disrupted, row.names = ENSG_Protein_Match_dMDT_distruptedPPI_canonical)

distances_df_NonDisrupted <- as.data.frame(dist_matrix_Nondisrupted, row.names = ENSG_Protein_Match_dMDT_NondistruptedPPI_canonical)

distances_df_random <- as.data.frame(random_paths, row.names = random_proteins)


#colnames(distances_df_Disrupted) <- cancer_census
#
#colnames(distances_df_NonDisrupted) <- cancer_census
#
#colnames(distances_df_random) <- cancer_census
```

```{r visualize_ccg}
library(ggplot2)
library(tidyverse)
# Assuming distances_df is the dataframe containing the distances
# Convert row names to a column

distances_long_random <- tibble::rownames_to_column(distances_df_random, "Protein")
distances_long_random_long <- distances_long_random %>% 
  gather(key = "Variable", value = "distance", -Protein)
distances_long_random_long$Category <- 'Random'

distances_long_disrupted <- tibble::rownames_to_column(distances_df_Disrupted, "Protein")
distances_long_disrupted_long <- distances_long_disrupted %>% 
  gather(key = "Variable", value = "distance", -Protein)
distances_long_disrupted_long$Category <- 'Disrupted'

distances_long_NonDisrupted <- tibble::rownames_to_column(distances_df_NonDisrupted, "Protein")
distances_long_NonDisrupted_long <- distances_long_NonDisrupted %>% 
  gather(key = "Variable", value = "distance", -Protein)
distances_long_NonDisrupted_long$Category <- 'NonDisrupted'


# Optionally, you can remove any rows with NA values (if any)
distances_long_disrupted_long <- distances_long_disrupted_long[complete.cases(distances_long_disrupted_long), ]
distances_long_NonDisrupted_long <- distances_long_NonDisrupted_long[complete.cases(distances_long_NonDisrupted_long), ]

distances_long <- rbind(distances_long_disrupted_long, distances_long_NonDisrupted_long, distances_long_random_long)
distances_long_uniq <- distances_long %>% dplyr::distinct()
```

```{r get_shortes}
distances_long_uniq_shortest <-  distances_long_uniq %>%
  group_by(Protein, Category) %>%
  summarise(min_distance = min(distance))

distances_long_uniq_shortest_rel <- distances_long_uniq %>%
    group_by(Category, distance) %>%
    summarise(count = n()) %>%
    ungroup() %>%
    group_by(Category) %>%
    mutate(percentage = count/sum(count))

distances_long_uniq_shortest_rel2_min <- distances_long_uniq_shortest %>% ungroup() %>%
    group_by(Category, min_distance) %>%
    summarise(count = n()) %>%
    ungroup() %>%
    group_by(Category) %>%
    mutate(percentage = count/sum(count))
```

```{r visualize_shortest_path}
# Create the plot
# Load the necessary library
library(ggplot2)

# Create the plot
ggplot(distances_long_uniq_shortest, aes(x = as.factor(min_distance), fill = Category)) + 
  geom_bar(position = "dodge") +
  labs(x = "Minimum Distance", y = "Count", fill = "Category") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) # Rotate x-axis labels for better visibility

my_palette_ccg <- c("#7ecc92",
"#22b948",
"#07661f")

# Create the plot
ggplot(distances_long_uniq_shortest_rel, aes(x = as.factor(distance) ,  y=percentage, fill = Category)) + 
  geom_bar(position = "dodge", stat='identity') +
  labs(x = "Minimum Distance", y = "Relative Frequency", fill = "Category") +
  scale_fill_manual(values = my_palette_ccg) + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + theme_classic()  + scale_fill_brewer(palette="Paired") # Rotate x-axis labels for better visibility

ggplot(distances_long_uniq_shortest_rel2_min, aes(x = as.factor(min_distance) ,  y=percentage, fill = Category)) + 
  geom_bar(position = "dodge", stat='identity') +
  labs(x = "Minimum Distance", y = "Relative Frequency", fill = "Category") +
  scale_fill_manual(values = my_palette_ccg) + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + theme_classic()  + scale_fill_brewer(palette="Paired")

## boxplot
distances_long_uniq_shortest_rel$Category <- factor(distances_long_uniq_shortest_rel$Category, levels = unique(distances_long_uniq_shortest_rel$Category))


p_ccg <- ggplot(distances_long_uniq_shortest_rel, aes(x = Category, y = distance, color = Category)) +
    geom_boxplot() +
    scale_color_manual(values = my_palette_ccg)  # use the color palette

p_ccg + theme_classic()  + scale_fill_brewer(palette="Paired")



p_ccg2 <- ggplot(distances_long_uniq_shortest_rel2_min, aes(x = Category, y = min_distance, color = Category)) +
    geom_boxplot() +
    scale_color_manual(values = my_palette_ccg)  # use the color palette

p_ccg2 + theme_classic()  + scale_fill_brewer(palette="Paired")
```


## ClinVar Genes
```{r ClinVar}
OncoKB_CancerList <- read.csv('/Users/tulaykarakulak/Documents/PhD/Projects/TumorProfiler/Analysis/May2024/cancerGeneList.tsv', sep='\t', header = TRUE)

head(OncoKB_CancerList)
```

```{r clinvar2}
ENSG_Protein_Match_oncoKB <- merge(OncoKB_CancerList, ensg_enst_ensp_pdb_pfams[, c(1,2,3,10)], by.x='Hugo.Symbol', 'Gene.name')

ENSG_Protein_Match_oncoKB_canonical <- unique(ENSG_Protein_Match_oncoKB[ENSG_Protein_Match_oncoKB$Protein.stable.ID %in% string_400PPI$protein1, 'Protein.stable.ID'])


dist_matrix_disrupted <- distances(graph, v = ENSG_Protein_Match_dMDT_distruptedPPI_canonical, to = ENSG_Protein_Match_oncoKB_canonical) 
dist_matrix_Nondisrupted <- distances(graph, v = ENSG_Protein_Match_dMDT_NondistruptedPPI_canonical_stringavailable, to = ENSG_Protein_Match_oncoKB_canonical)

## Random DATA
# Create a random set of proteins
set.seed(123)  # For reproducibility
#random_proteins <- sample(V(graph)$name, 500)

# Calculate shortest paths
random_paths <- distances(graph, v = random_proteins, to = ENSG_Protein_Match_oncoKB_canonical)

distances_df_Disrupted <- as.data.frame(dist_matrix_disrupted, row.names = ENSG_Protein_Match_dMDT_distruptedPPI_canonical)

distances_df_NonDisrupted <- as.data.frame(dist_matrix_Nondisrupted, row.names = ENSG_Protein_Match_dMDT_NondistruptedPPI_canonical)

distances_df_random <- as.data.frame(random_paths, row.names = random_proteins)


distances_long_random <- tibble::rownames_to_column(distances_df_random, "Protein")
distances_long_random_long <- distances_long_random %>% 
  gather(key = "Variable", value = "distance", -Protein)
distances_long_random_long$Category <- 'Random'

distances_long_disrupted <- tibble::rownames_to_column(distances_df_Disrupted, "Protein")
distances_long_disrupted_long <- distances_long_disrupted %>% 
  gather(key = "Variable", value = "distance", -Protein)
distances_long_disrupted_long$Category <- 'Disrupted'

distances_long_NonDisrupted <- tibble::rownames_to_column(distances_df_NonDisrupted, "Protein")
distances_long_NonDisrupted_long <- distances_long_NonDisrupted %>% 
  gather(key = "Variable", value = "distance", -Protein)
distances_long_NonDisrupted_long$Category <- 'NonDisrupted'

# Optionally, you can remove any rows with NA values (if any)
distances_long_disrupted_long <- distances_long_disrupted_long[complete.cases(distances_long_disrupted_long), ]
distances_long_NonDisrupted_long <- distances_long_NonDisrupted_long[complete.cases(distances_long_NonDisrupted_long), ]

distances_long <- rbind(distances_long_disrupted_long, distances_long_NonDisrupted_long, distances_long_random_long)

distances_long_uniq <- distances_long %>% dplyr::distinct()


distances_long_uniq_shortest <-  distances_long_uniq %>%
  group_by(Protein, Category) %>%
  summarise(min_distance = min(distance))

distances_long_uniq_shortest_rel2_min <- distances_long_uniq_shortest %>% ungroup() %>%
    group_by(Category, min_distance) %>%
    summarise(count = n()) %>%
    ungroup() %>%
    group_by(Category) %>%
    mutate(percentage = count/sum(count))

ggplot(distances_long_uniq_shortest_rel2_min, aes(x = as.factor(min_distance) ,  y=percentage, fill = Category)) + 
  geom_bar(position = "dodge", stat='identity') +
  labs(x = "Minimum Distance", y = "Relative Frequency", fill = "Category") +
  scale_fill_manual(values = my_palette_ccg) + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + theme_classic()   + scale_fill_brewer(palette="Paired")
```



